src/main/resources/data/minecoprocessors/books/manual.txt

```diff
@@ -1,130 +1,291 @@
+
+ §1MOV a, 12§r
+ §1MOV b, 10§r
+ §1SUB a, b§r
+
+will result in register §1a§r containing §12§r as present value.
+
+~~~
+Comments start with a semicolon:
+
+ §1; This is a comment§r
+
+Labels allow to name program addresses. They are used for conditions, loops, or functions:
+
+§4start:§r
+ §1JMP§r §4start§r
+~~~
+Numbers can be hex, decimal, octal, as well as binary. A byte with all bits set (§11§r) can be written as:
+
+Hexadecimal: §10xff§r
+Decimal: §1-1§r
+Octal: §10o377§r
+Binary: §111111111b§r
+
+The decimal value above is equal to §1-1§r because the byte is ...
+~~~
+interpreted as signed value. That means the highest bit §110000000b§r adds §1-128§r to all other bits, which add §11§r, §12§r, §14§r, §18§r, §116§r, §132§r, and §164§r to the value.
+
+In most cases, binary or hexadecimal representations are best, as we are often dealing with the bits in registers.
+~~~
+§nCore Registers§r
+
+The CPU has four one byte registers §1a, b, c,§r and §1d§r to store values and calculate.
+
+These "variables" do not change any of the signals outside the CPU - they are only internal.
 ~~~
-§1Instructions§r
+§nPorts and Port Config§r
 
-§lMOV a, b§r Move
-§lPUSH a§r Push to Stack
-§lPOP a§r Pop from Stack
-§lNOP§r No Operation
-§lHLT§r Halt Processor
-§lCLC§r Clear Carry
-§lSEC§r Set Carry
-§lCLZ§r Clear Zero
-§lSEZ§r Set Zero
-§lHLT§r Halt Processor
-§lWFE§r Sleep (experimental)
+The CPU interacts with the world around via special registers. These are mapped to the four sides of the device, and we need to tell the CPU if we like to read signals from a side, or want to output redstone power there.
+
+Additionally, we can
 ~~~
-§1Arithmetic Instructions§r
+specify for each I/O if we are interested in the numeric "strength" value of the signal, or only the on/off data.
+§oAlso known as §1analog§r and §1digital§r mode.§r
+
+Lastly, ports can also be used to reset the controller by leaving a side unconfigured or "unconfiguring" it explicitly.
 
-§lADD a, b§r Add
-§lSUB a, b§r Subtract
-§lMUL a§r Multiply
-§lDIV a§r Divide
-§lINC a§r Increment by 1
-§lDEC a§r Decrement by 1
 ~~~
-§1Logical Instructions§r
+Signals are accessible primarily through the registers §1pf pb pl pr§r:
+
+§80001b | §1pf§r : front
+§80010b | §1pb§r : back
+§80100b | §1pl§r  : left
+§81000b | §1pr§r : right
 
-§lAND a, b§r Bitwise AND
-§lOR a, b§r Bitwise OR
-§lXOR a, b§r Bitwise XOR
-§lNOT a§r Bitwise NOT
+Gray annotated are the binary bit values to configure the I/Os in the §1ports§r and §1adc§r registers.
 ~~~
-§1Shift Instructions§r
+The first instruction in a listing is normally to write the register §1ports§r, where cleared bit configure outputs and set bits inputs. E.g.
 
-§lSHL a§r Shift Left
-§lSHR a§r Shift Right
-§lSAR a§r Arithmetic Right
-§lSAL a§r Arithmetic Left
-§lROR a§r Rotate Right
-§lROL a§r Rotate Right
+  §1MOV ports, 1010b§r
+
+defines §1pf§r and §1pl§r as outputs, §1pb§r and §1pr§r as inputs. To use an I/O for reset, set its bit
```

Comment:
Consider changing `To use an I/O for reset,` to `To use an I/O port for reset,` for clarity and consistiency